### DISCUSSION TITLE
Access-Control-Allow-Headers editable from the general config for graphql

### DISCUSSION THREAD

**Discussion by martijnvanonz** [craftcms/cms#10406] [Ideas] [answered]
It would be nice if you can change Access-Control-Allow-Headers from the general config as you can do for allowedGraphqlOrigins. We use custom headers for certain queries. This now results in cors errors.

**Comment by estepnv** [comment]
up

**Comment by brandonkelly** [comment]
Why not just set the `Access-Control-Allow-Headers` header from the same controller action where you’re adding the custom header? Seems like it would be a little more maintainable that way.

**Comment by timkelty** [comment, marked as answer]
@martijnvanonz @estepnv This is possible in the upcoming 4.11/5.3 release via `config/app.php`. See https://github.com/craftcms/cms/pull/15397 for examples and details.
